internal/util/type_conversion.go

```diff
@@ -30,6 +34,9 @@ func ToOtelValue(value interface{}) interface{} {
 	case float32:
 		return float64(v)
 	case float64:
+		if math.IsNaN(v) || math.IsInf(v, 0) {
+			return nil
+		}
```

Comment:
I am fine with this, but just wanted to bring it up for discussion.
If OTEL is okay with NaN and INF than we shouldn't drop it as this point.

The thing that doesn't support NaN and INF is the backend, so the exporter should be the place that decides to drop it.

Functionally its the same though.

Reply:
Sure I can remove this check. It should get dropped at the distribution step, but that'll lead to sending unnecessary metrics through our pipeline.